File: Back/Back/Models/DetalleCompra.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Back.Models
{
    public class DetalleCompra
    {
        public int Id { get; set; }
        public int ProductoId { get; set; }
        public Producto Producto { get; set; }

        public int Cantidad { get; set; }
        public float PrecioUnitario { get; set; }
        public float Subtotal { get; set; }
        public int LocalId { get; set; }
        public Local Local { get; set; }

        public string CompradorId { get; set; }

    }
}

File: Front/src/app/interfaces/detalle-compra.ts
import { Local } from "./local";
import { Producto } from "./producto";

export interface DetalleCompra {
    id?: number; // Hacer el campo id opcional con el operador "?"
    productoId: string;
    cantidad: number;
    precioUnitario: number;
    localId: number; 
    compradorId: string; 
    subtotal: number; 
    producto?: Producto; // Agregar referencia al producto
    local?: Local; // Agregar referencia al local
    foto?: string; 
}
